### PR TITLE
Auto-generation fixtures

### PR DESCRIPTION
**Pull request by m4dz**
This PR allow defining fixtures as a model for [Mockaroo](https://www.mockaroo.com) service and then auto-generate and import fixtures on the fly. It adds two new floags to the CLI:

- `-g` to enable the auto-generation service
- `-n` to specify the number of records wanted

Note that it need an API key (freely available) available in the MOCKAROO_API_KEY env variable.

### CODE REVIEW

**Comment by frankrousseau** [comment]
@jsilvestre Can I merge this PR?

File: main.coffee [hunk 1]
```diff
@@ -303,7 +314,36 @@ class FixtureManager
...
+                        client
+                            .generate
+                                count: @counter
+                                fields: data
+                            .then (records) ->
+                                callback null, records
+                            .catch (err) ->
+                                if err instanceof Mockaroo.errors.InvalidApiKeyError
+                                    err = 'invalid api key'
+                                else if err instanceof Mockaroo.errors.UsageLimitExceededError
```
Comment (frankrousseau) [inline main.coffee]: Are you sure that this line is less than 80 chars wide?

**Comment by m4dz** [comment]
Ok, so tests are passing locally on a cozy base instance, and fail when passing on a freshly resetted db (w/ `cozy-dev db:reset`). I suppose that Travis fails for the same reasons but I can't find where's the problem (and my knowledge about the DS and Couch are a limit for me at this point). @poupotte, @jsilvestre, @aenario, any thoughs ?

**Comment by frankrousseau** [comment]
Why did you rename `tests` in `test`? It's `tests` in every Cozy module.

**Comment by m4dz** [comment]
I renamed it to `test` because it's a convention across all testing tools that the tests folder is named `test`. After discussing about it with @jsilvestre and [debate about it's relevance](https://twitter.com/m4d_z/status/632111832203857920), we admit it's a mistake and we choose to rename it to `test`, and piecemeal change the folder name to `test` in cozy apps :grinning:.

**Comment by aenario** [comment]
Mockaroo use babel, babel use core-js, core-js break chai https://github.com/zloirock/core-js/issues/89

**Comment by m4dz** [comment]
@aenario I don't see what you mean… broken tests on a resetted cozy db instance aren't due to the mockaroo addition : they'll break on the master branch too :disappointed:. I haven't any issue running the auto-generation fixtures.

**Comment by frankrousseau** [comment]
Ok, `test` is ugly. But if it's a standard... (sad)

About your bug @m4dz it's probably due to requests that are not declared (if they occur only on empty database).

**Comment by m4dz** [comment]
> Ok, test is ugly. But if it's a standard... (sad)

I agree, I'm totally convinced by the theory of the "s" defective key :laughing:.

> it's probably due to requests that are not declared

Yep. I'll try to fix it today.

**Comment by m4dz** [comment]
Ok, so:

1. tests fail locally on master branch only if they're applied on another db than `cozy` (some tests assume the tested couch db is the `cozy` one). My first mistake.
2. tests are performed on the `main.js` file (not the coffee version) when this one is present, so I had som inconsistencies across same tests set because it wasn't compiled each time. Second mistake.
3. @aenario, you're true, and I don't firstly understand what you mean. Core-js breaks Number prototype constructor and then chai assetions base on numbers fail (miserably). Third mistake.

So, now we have a culprit, i'll fix the assertions :expressionless: to bypass the issue…

**Comment by m4dz** [comment]
As well as the issue is fixed in the core-js 1.0.0 branch, I've just submitted a patch to Mockaroo to update their deps as well : https://github.com/mockaroo/mockaroo-node/pull/1.

**Comment by clochix** [comment]
So, what's the status of this PR ? @m4dz, @jsilvestre ?

**Comment by m4dz** [comment]
The code is fully functional, but the tests always fail due to the bug mentioned by @aenario, and the PR on Mockaroo is still alive.

I suggest we could merge this one (even with the tests failing) as we use it intensively (eg on Contacts for now), and I will up my PR on Mockaroo to fix the bug :)

**Pull request merged**